src/mlpack/core/util/split_data.hpp

```diff
@@ -0,0 +1,155 @@
+#ifndef __MLPACK_CORE_UTIL_SPLIT_DATA_HPP
+#define __MLPACK_CORE_UTIL_SPLIT_DATA_HPP
+
+#include <mlpack/core.hpp>
+
+#include <iterator>
+#include <numeric>
+#include <random>
+#include <tuple>
+#include <vector>
+
+namespace mlpack {
+namespace util {
+
+/**
+ *Split training data and test data, please define
+ *ARMA_USE_CXX11 to enable move of c++11
+ */
+class TrainTestSplit
+{
+public:
+  /**
+   * @brief TrainTestSplit
+   * @param testRatio the ratio of test data
+   * @param seed seed of the random device
+   * @warning slice should not less than 1
+   */
+  TrainTestSplit(double testRatio,                
+                 arma::arma_rng::seed_type seed = 0) :
+    seed(seed),    
+    testRatio(testRatio)
+  {    
+  }
+
+  /**
+   *Split training data and test data, please define
+   *ARMA_USE_CXX11 to enable move of c++11
+   *@param input input data want to split
+   *@param label input label want to split
+   *@param testRatio the ratio of test data
+   *@param seed seed of the random device
+   *@code
+   *arma::mat input = loadData();
+   *arma::Row<size_t> label = loadLabel();
+   *arma::mat trainData;
+   *arma::mat testData;
+   *arma::Row<size_t> trainLabel;
+   *arma::Row<size_t> testLabel;
+   *std::random_device rd;
+   *TrainTestSplit tts(0.25);
+   *tts.Split(input, label, trainData, testData, trainLabel,
+   *          testLabel);
+   *@endcode
+   */
+  template<typename T, typename U>
+  void Split(arma::Mat<T> const &input,
+             arma::Row<U> const &inputLabel,
+             arma::Mat<T> &trainData,
+             arma::Mat<T> &testData,
+             arma::Row<U> &trainLabel,
+             arma::Row<U> &testLabel)
+  {
+    size_t const testSize =
+        static_cast<size_t>(input.n_cols * testRatio);
+    size_t const trainSize = input.n_cols - testSize;
+    trainData.set_size(input.n_rows, trainSize);
+    testData.set_size(input.n_rows, testSize);
+    trainLabel.set_size(trainSize);
+    testLabel.set_size(testSize);
+
+    using Col = arma::Col<size_t>;
+    arma::arma_rng::set_seed(seed);
```

Comment:
I don't think there's any need for this, we should just assume that the user has called `math::RandomSeed()` appropriately.